File: golang/3010.divide-an-array-into-subarrays-with-minimum-cost-i.go
func minimumCost(nums []int) int {

	minA, minB := nums[1], nums[2]

	if minA > minB {
		minA, minB = minB, minA
	}

	for i := 3; i < len(nums); i++ {
		if minA > nums[i] {
			minB, minA = minA, nums[i]
		} else if minA == nums[i] || minB > nums[i] {
			minB = nums[i]
		}
	}

	return nums[0] + minA + minB
}

File: cpp/3010.divide-an-array-into-subarrays-with-minimum-cost-i.cpp
#include <utility>
#include <vector>
using std::vector;

class Solution {
public:
  int minimumCost(vector<int> &nums) {
    int minA = nums[1], minB = nums[2];
    if (minA > minB) {
      std::swap(minA, minB);
    }

    for (int i = 3; i < nums.size(); i++) {
      if (nums[i] < minA) {
        minB = minA;
        minA = nums[i];
      } else if (nums[i] == minA || nums[i] < minB) {
        minB = nums[i];
      }
    }

    return nums[0] + minA + minB;
  }
};